Library/Homebrew/test/cmd/uses_spec.rb

```diff
@@ -44,4 +45,22 @@
       .and not_to_output.to_stderr
       .and be_a_success
   end
+
+  it "handles unavailable formula", :integration_test do
+    setup_test_formula "foo"
+    setup_test_formula "bar"
+    setup_test_formula "optional", <<~RUBY
+      url "https://brew.sh/optional-1.0"
+      depends_on "bar" => :optional
+    RUBY
+
+    expect_any_instance_of(::Homebrew::CLI::NamedArgs)
```

Comment:
Alas, I couldn't figure out how to induce `FormulaUnavailableError` organically, so I used an expectation here.

Additionally, it doesn't hook when using `brew` in a test, so I created a command instance and directly invoked `run`.